providers/base/fetch.go

```diff
@@ -126,21 +129,27 @@ func (p *Provider[K, V]) startWebSocket(ctx context.Context, subIDs []K, respons
 	return func() error {
 		// Start the websocket query handler. If the connection fails to start, then the query handler
 		// will be restarted after a timeout.
+		restarts := 0
 		for {
 			select {
 			case <-ctx.Done():
-				p.logger.Info("provider stopped via context")
+				p.logger.Info("web socket stopped via context")
 				return ctx.Err()
 			default:
+				if restarts > 0 {
+					p.logger.Debug("restarting websocket query handler", zap.Int("num_restarts", restarts))
+
+					// If the websocket query handler returns, then the connection was closed. Wait for
+					// a bit before trying to reconnect.
+					time.Sleep(p.wsCfg.ReconnectionTimeout)
+				}
```

Comment:
This should speed up shutdown time rn which otherwise would trigger the reconnection sleep.